File: Notifications/src/main/java/com/example/entities/Appointment.java
package com.example.entities;

import com.example.enums.EAppointmentStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalTime;
import java.util.Date;
import java.util.UUID;

@Entity
@Table(name = "appointment")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Appointment {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "uuid", columnDefinition = "BINARY(16)")
    private UUID id;
    private Date date;
    private LocalTime time;
    @Enumerated(EnumType.STRING)
    private EAppointmentStatus appointmentStatus;
    @ManyToOne
    @JoinColumn(nullable = false, name = "doctor_id")
    private Doctor doctor;
    @ManyToOne
    @JoinColumn(nullable = false, name = "patient_id")
    private Patient patient;

}

File: GUI/src/app/shared/models/appointment/appointment.ts
import { v4 as uuidv4 } from 'uuid';
import { Doctor } from '../users/doctor/doctor';
import { Patient } from '../users/patient/patient';

export class Appointment {
    public id: string;
    date: string;
    time: string;
    appointmentStatus: string;
    doctor: Doctor
    patient: Patient

  constructor() {
    this.id = uuidv4();
  }
}
